File: src/main/java/com/example/employee/controller/UserController.java
//this is a place where crud operations are performed
//these are the rest API's
package com.example.employee.controller;


import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

import com.example.employee.converter.UserConverter;
import com.example.employee.dto.UserDto;
import com.example.employee.service.UserService;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

//it maps the url 
@RequestMapping("/user")
// anotation for saying that its a rest API
@RestController

public class UserController{

    @Autowired
    private UserService userService;

    //get all the users
    @RequestMapping("/getAllUsers")
    public List<UserDto> getAllUsers(){
        return userService.getAllUsers().stream().map(UserConverter::entityToDto).collect(Collectors.toList());
    }

    //UserConverter.entityToDto(userRepository.getOne(userId));
//services take from entities and controller takes from dto, so need to convert the entities to dto.
    //get users by id
    @RequestMapping("/getUser/{userId}")
    public UserDto getUserById(@PathVariable Integer userId){
        return UserConverter.entityToDto(userService.getUserById(userId));
    }

}

File: src/main/resources/static/js/UserController.js
var module = angular.module('employee.controllers', []);
module.controller("UserController", ["$scope", "UserService",
    function ($scope, UserService) {
     $scope.allUsers = [];
     $scope.currentUserId = 0;
     $scope.fetchUser = () => {
        UserService.getUserById($scope.currentUserId).then((user) => {
            $scope.UserById=user.data;  
            console.log(user.data); 
        });
     }
     //get all the users, using service interface
     // then(users)-> its a function called as calledback , users is the result that we get from getALLusers()
     //or can even use await fetch()


    /*
    * users = await UserService.getAllUsers();
    * $scope.users = users.data;
    */

        UserService.getAllUsers().then((users) => {
            $scope.allUsers = users.data;
          //calling the service methid
        });
    }]);